File: Demo/NUITests/UIKitTests/UITabBar/NUITabBarRendererBackgroundImageTests.m
//
//  NUITabBarRendererBackgroundImageTests.m
//  NUIDemo
//
//  Created by Paul Williamson on 03/06/2014.
//  Copyright (c) 2014 Tom Benner. All rights reserved.
//

#import <XCTest/XCTest.h>

#import "NUIConstants.h"
#import "NUIFileMonitor.h"
#import "UIView+NUI.h"

#import "NUIDemo-Swift.h"
#import <FBSnapshotTestCase/UIImage+Compare.h>

@interface NUITabBarRendererBackgroundImageTests : XCTestCase
@property (strong, nonatomic) UITabBar *sut;
@end

@implementation NUITabBarRendererBackgroundImageTests

- (void)setUp
{
    [super setUp];
    [NUISettings initWithStylesheet:@"TestTheme.NUI"];
    
    _sut = [[UITabBar alloc] initWithFrame:CGRectMake(0, 0, 320, 44)];
    _sut.nuiClass = @"TabBarWithImages";
    [_sut applyNUI];
}

- (void)tearDown
{
    _sut = nil;
    [super tearDown];
}

// background-image (Image)
- (void)testSetBackgroundImage
{
    UIImage *image = [UIImage imageNamed:@"background_image"];
    UIImage *backgroundImage = _sut.backgroundImage;
    XCTAssertTrue([backgroundImage fb_compareWithImage:image tolerance:0], @"NUI should set tab bar background image");
}

// background-image-insets (Box)
- (void)testSetBackgroundImageInsets
{
    UIImage *backgroundImage = _sut.backgroundImage;
    UIEdgeInsets insets = UIEdgeInsetsMake(2, 2, 2, 2);
    XCTAssertTrue(UIEdgeInsetsEqualToEdgeInsets(backgroundImage.capInsets, insets), @"NUI should set tab bar background image insets");
}

// selected-image (Image)
- (void)testSetSelectedImage
{
    UIImage *image = [UIImage imageNamed:@"foreground_image"];
    UIImage *selectedImage = _sut.selectionIndicatorImage;
    XCTAssertTrue([selectedImage fb_compareWithImage:image tolerance:0], @"NUI should set tab bar selected image");
}

@end

File: Demo/NUITests/UIKitTests/UITabBar/NUITabBarRendererBackgroundImageTests.swift
//
//  NUITabBarRendererBackgroundImageTests.swift
//  NUIDemo
//
//  Created by Sergej Jaskiewicz on 29.08.16.
//  Copyright © 2016 Tom Benner. All rights reserved.
//

import XCTest
import FBSnapshotTestCase
@testable import NUIDemo

class NUITabBarRendererBackgroundImageTests: XCTestCase {

    var sut: UITabBar!

    override func setUp() {
        super.setUp()

        NUISettings.initWithStylesheet("TestTheme.NUI")
        sut = UITabBar(frame: CGRect(x: 0, y: 0, width: 320, height: 44))
        sut.nuiClass = "TabBarWithImages"
        sut.applyNUI()
    }

    // background-image (Image)
    func testSetBackgroundImage() {
        let backgroundImage = sut.backgroundImage
        XCTAssertTrue(backgroundImage!.fb_compare(with: #imageLiteral(resourceName: "background_image.png"), tolerance: 0),
                      "NUI should set tab bar background image")
    }

    // background-image-insets (Box)
    func testSetBackgroundImageInsets() {
        let backgroundImage = sut.backgroundImage
        let insets = UIEdgeInsets(top: 2, left: 2, bottom: 2, right: 2)
        XCTAssertEqual(backgroundImage?.capInsets, insets, "NUI should set tab bar background image insets")
    }

    // selected-image (Image)
    func testSetSelectedImage() {
        let selectedImage = sut.selectionIndicatorImage
        XCTAssertTrue(selectedImage!.fb_compare(with: #imageLiteral(resourceName: "foreground_image.png"), tolerance: 0), "NUI should set tab bar selected image")
    }
}
